compiler/vm/vmgen.nim

```diff
@@ -2683,6 +2683,8 @@ proc gen(c: var TCtx; n: PNode; dest: var TDest; flags: TGenFlags = {}) =
       genCastIntFloat(c, n, dest)
   of nkType:
     genTypeLit(c, n.typ, dest)
+  of nkError:
+    assert false
```

Comment:
```suggestion
```
An `nkError` should not be able to reach here. If one does, then there's a bug with `mirgen`.

Reply:
I should make that an internal error really.

In the future i should have a const that's, `nodeKindsAllowedForVMGen` or something, and then guard it all and all case statements are exhaustive.